subprojects/core/src/main/java/org/gradle/api/internal/tasks/execution/ExecuteActionsTaskExecuter.java

```diff
@@ -192,12 +191,12 @@ public String getIdentity() {
         }
 
         @Override
-        public ExecutionOutcome execute(IncrementalChangesContext context) {
+        public ExecutionOutcome execute(Optional<ExecutionStateChanges> changes) {
```

Comment:
We could pass `IncrementalChangesInternal` here, no?

Reply:
We could also have two methods instead of passing an `Optional`.

Reply:
I am now passing `InputChangesInternal`.

Regarding the two methods, I don't think this will make it easier for the one implementing the unit of work. How did you imagine that to work?